core/src/main/java/org/apache/accumulo/core/client/admin/NewTableConfiguration.java

```diff
@@ -146,4 +163,116 @@ public NewTableConfiguration enableSummarization(SummarizerConfiguration... conf
+      Set<Text> colFams = entry.getValue();
+      String value = LocalityGroupUtil.encodeColumnFamilies(colFams);
+      localityProps.put(Property.TABLE_LOCALITY_GROUP_PREFIX + entry.getKey(), value);
+    }
+    // localityProps.put(Property.TABLE_LOCALITY_GROUPS.getKey(), Joiner.on(",").join(groups.keySet()));
+    localityProps.put(Property.TABLE_LOCALITY_GROUPS.getKey(), groups.keySet().stream().collect(Collectors.joining(",")));
+    // localityProps.put(Property.TABLE_LOCALITY_GROUPS.getKey(), Stream.of(groups.keySet().collect(Collectors.joining(","));
+    // Stream.of(groups.keySet()).collect(joining(","));
+  }
+
+  /**
+   * Configure iterator settings for a table prior to its creation.
+   *
+   * Additional calls to this method before table creation will overwrite previous iterator settings.
+   *
+   * @param setting
+   *          object specifying the properties of the iterator
+   * @throws AccumuloSecurityException
+   *           thrown if the user does not have the ability to set properties on the table
+   * @throws AccumuloException
+   *           if a general error occurs
+   * @throws TableNotFoundException
+   *           if the table does not exist
+   *
+   * @since 2.0.0
+   */
+  public void attachIterator(IteratorSetting setting) throws AccumuloException, TableNotFoundException {
+    attachIterator(setting, EnumSet.allOf(IteratorScope.class));
+  }
+
+  /**
+   * Configure iterator settings for a table prior to its creation.
+   *
+   * @param setting
+   *          object specifying the properties of the iterator
+   * @param scopes
+   *          enumerated set of iterator scopes
+   * @throws AccumuloException
+   *           if a general error occurs
+   * @throws AccumuloSecurityException
+   *           thrown if the user does not have the ability to set properties on the table
+   * @throws TableNotFoundException
+   *           if the table does not exist
+   *
+   * @since 2.0.0
+   */
+  public void attachIterator(IteratorSetting setting, EnumSet<IteratorScope> scopes) throws AccumuloException, TableNotFoundException {
+    checkArgument(setting != null, "setting is null");
+    checkArgument(scopes != null, "scopes is null");
+    if (iteratorProps.isEmpty()) {
+      iteratorProps = new HashMap<>();
+    }
+    checkIteratorConflicts(setting, scopes);
+    for (IteratorScope scope : scopes) {
+      String root = String.format("%s%s.%s", Property.TABLE_ITERATOR_PREFIX, scope.name().toLowerCase(), setting.getName());
+      for (Entry<String,String> prop : setting.getOptions().entrySet()) {
+        iteratorProps.put(root + ".opt." + prop.getKey(), prop.getValue());
+      }
+      iteratorProps.put(root, setting.getPriority() + "," + setting.getIteratorClass());
+    }
+  }
+
+  private void checkIteratorConflicts(IteratorSetting setting, EnumSet<IteratorScope> scopes) throws AccumuloException, TableNotFoundException {
+    checkArgument(setting != null, "setting is null");
```

Comment:
Could the code in TableOperationsHelper be refactored such it could be called from here.  I am thinking of the following refactoring.

```java
class TableOperationsHelper {
  //add this method and call it from NewTableConfig
  public void checkIteratorConflicts(Map<String, String> props, IteratorSetting setting, EnumSet<IteratorScope> scopes) throws AccumuloException, TableNotFoundException {
 }
  //refactor this existing method to call above method
  public void checkIteratorConflicts(String tablename, IteratorSetting setting, EnumSet<IteratorScope> scopes) throws AccumuloException, TableNotFoundException {
 }
}
```

Reply:
@keith-turner I have a question concerning this. It's strightforward enough to create the new method and refactor the old. Perhaps I'm not completely versed on its usage yet, but wouldn't calling the method from NewTableConfiguration require instantiation of the TableOperationsImpl object? When I was playing around with it, this required a ClientConfiguration object which then required an Instance object,etc. It seemed to get messy attempting the call. Am I missing something obvious?

Reply:
There is at least one more very similar instance of checkIteratorConflicts for namespaces as well. Would it make sense to create a new ticket to consolidate the various permutations of this method and perhaps place the new versions into  the iteratorUtil class where it could more easily be called by the various classes that need that check?

Reply:
Can you make the new method in TableOpHelper static?

Reply:
I'll give it a shot and see how that works.

Reply:
Created static version of checkIteratorConflicts and refactored existing version of method within TableOptionsHelper to make use of static version. Used static version to check conflicts in new NewTableConfiguration methods.